src/Network/Freddy.hs

```diff
@@ -90,11 +93,13 @@ deliverWithResponse connection request = do
     AMQP.msgExpiration    = Request.expirationInMs request
   }
 
-  AMQP.publishMsg' (amqpChannel connection) "" (Request.queueName request) True msg
-  AMQP.publishMsg (amqpChannel connection) topicExchange (Request.queueName request) msg
+  responseChannelListener <- BC.newBChanListener $ eventChannel connection
+
+  AMQP.publishMsg' (amqpProduceChannel connection) "" (Request.queueName request) True msg
+  AMQP.publishMsg (amqpProduceChannel connection) topicExchange (Request.queueName request) msg
 
   amqpResponse <- timeout (Request.timeoutInMicroseconds request) $
-    waitForResponse (responseChannelListener connection) correlationId $ matchingCorrelationId correlationId
+    waitForResponse responseChannelListener correlationId $ matchingCorrelationId correlationId
```

Comment:
This could use some local vars / `where` / `let` bindings to name parts of this big expression